core/commands/repo.go

```diff
@@ -19,10 +21,17 @@ var RepoCmd = &cmds.Command{
 	},
 
 	Subcommands: map[string]*cmds.Command{
-		"gc": repoGcCmd,
+		"gc":   repoGcCmd,
+		"stat": repoStatCmd,
 	},
 }
 
+type RepoStat struct {
+	localPath string
+	szBytes   uint64
+	szBlocks  uint64
```

Comment:
This `szBlocks` / `numBlocks` could be used for gc constraint, if it takes less time to calculate than `GetStorageUsage`.